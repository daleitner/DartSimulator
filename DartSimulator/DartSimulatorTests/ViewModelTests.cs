using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Reporters;
using DartSimulator;
using DartSimulator.Controller;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Moq;
using Should;

namespace DartSimulatorTests
{
	[TestClass]
	[UseReporter(typeof(DiffReporter))]
	public class ViewModelTests
	{
		private readonly Mock<ISimulationController> controllerMock = new Mock<ISimulationController>();
		[TestMethod]
		public void WhenCreatingNewViewModel_LegsShouldBeTenThousand()
		{
			var viewModel = new MainViewModel(this.controllerMock.Object);
			viewModel.AmountLegs.ShouldEqual(10000);
		}

		[TestMethod]
		public void WhenClickSimulateButton_ThenControllerShouldGetTriggered()
		{
			var viewModel = new MainViewModel(this.controllerMock.Object);
			viewModel.StartCommand.Execute(null);
			this.controllerMock.Verify(x => x.StartSimulation(), Times.Once, "Start Simulation was not triggered");
		}

		[TestMethod]
		public void WhenStartSimulation_ThenAllPropertiesShouldBeFilled()
		{
			var result = new Result
			{
				Average = 50.5,
				BestLeg = 18,
				DartAverage = 27,
				DoubleQuote = 25.2,
				Hundrets = 7,
				HundretFourties = 3,
				HundretEighties = 1,
				WorstLeg = 36
			};
			this.controllerMock.Setup(x => x.StartSimulation()).Returns(result);
			var viewModel = new MainViewModel(this.controllerMock.Object);
			viewModel.StartCommand.Execute(null);
			Approvals.Verify(GetResultProperties(viewModel));
		}

		private string GetResultProperties(MainViewModel viewModel)
		{
			string ret = "";
			ret += "Anzahl Legs: " + viewModel.Legs.Count;
			ret += "\nSelectedLeg: " + viewModel.SelectedLeg;
			ret += "\n3-Dart-Average: " + viewModel.Average;
			ret += "\nDoppelquote: " + viewModel.SimulatedDoubleQuote;
			ret += "\nBest Leg: " + viewModel.BestLeg;
			ret += "\nWorst Leg: " + viewModel.WorstLeg;
			ret += "\nDart-Average: " + viewModel.DartAverage;
			ret += "\n100er: " + viewModel.Hundrets;
			ret += "\n140er: " + viewModel.HundretFourties;
			ret += "\n180er: " + viewModel.HundretEighties;
			return ret;
		}
	}
}
