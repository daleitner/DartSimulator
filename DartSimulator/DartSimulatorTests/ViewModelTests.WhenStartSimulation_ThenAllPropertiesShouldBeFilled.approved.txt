Anzahl Legs: 7
SelectedLeg: 
3-Dart-Average: 50,5
Doppelquote: 0% (7/30)
Best Leg: 18
Worst Leg: 36
Dart-Average: 27
100er: 7
140er: 3
180er: 1